modules/calib3d/src/hal_replacement.hpp

```diff
@@ -0,0 +1,132 @@
+// indirect, incidental, special, exemplary, or consequential damages
+// (including, but not limited to, procurement of substitute goods or services;
+// loss of use, data, or profits; or business interruption) however caused
+// and on any theory of liability, whether in contract, strict liability,
+// or tort (including negligence or otherwise) arising in any way out of
+// the use of this software, even if advised of the possibility of such damage.
+//
+//M*/
+
+#ifndef OPENCV_CALIB3D_HAL_REPLACEMENT_HPP
+#define OPENCV_CALIB3D_HAL_REPLACEMENT_HPP
+
+#include "opencv2/core/hal/interface.h"
+
+#if defined(__clang__)  // clang or MSVC clang
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wunused-parameter"
+#elif defined(_MSC_VER)
+#pragma warning(push)
+#pragma warning(disable : 4100)
+#elif defined(__GNUC__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+#endif
+
+//! @addtogroup calib3d_hal_interface
+//! @note Define your functions to override default implementations:
+//! @code
+//! #undef hal_add8u
+//! #define hal_add8u my_add8u
+//! @endcode
+//! @{
+
+/**
+   @brief Project points from 3D world space to 2D screen space using rotation and translation matrix and camera intrinsic parameters
+   @param src_data Pointer to source 3D points array
+   @param src_step Step between consecutive points in 3D points array
+   @param src_size Amount of points in 3D points array
+   @param dst_data Pointer to resulting 2D points array
+   @param dst_step Step between consecutive points in 2D resulting array
+   @param rt_data Pointer to 3x4 array containing rotation-then-translation matrix
+   @param intrinsics_data Pointer to camera intrinsic parameters vector containing [fx, fy, cx, cy, k0, k1, k2, k3]
+*/
+inline int hal_ni_projectpoints32f(const float* src_data, uint32_t src_step, uint32_t src_size, float* dst_data, uint32_t dst_step, const float* rt_data, const float* intrinsics_data) { return CV_HAL_ERROR_NOT_IMPLEMENTED; }
+
+//! @cond IGNORED
+#define cv_hal_projectpoints32f hal_ni_projectpoints32f
+//! @endcond
+
+/**
+   @brief Project points from 3D world space to 2D screen space using rotation and translation matrix and camera intrinsic parameters
+   @param src_data Pointer to source 3D points array
+   @param src_step Step between consecutive points in 3D points array
+   @param src_size Amount of points in 3D points array
+   @param dst_data Pointer to resulting 2D points array
+   @param dst_step Step between consecutive points in 2D resulting array
+   @param rt_data Pointer to 3x4 array containing rotation-then-translation matrix
+   @param intrinsics_data Pointer to camera intrinsic parameters vector containing [fx, fy, cx, cy, k0, k1, k2, k3]
+*/
+inline int hal_ni_projectpoints64f(const double* src_data, uint32_t src_step, uint32_t src_size, double* dst_data, uint32_t dst_step, const float* rt_data, const float* intrinsics_data) { return CV_HAL_ERROR_NOT_IMPLEMENTED; }
```

Comment:
same comments as above. We need camera model.

Also, do we really need 64f flavor? Especially given that `rt_data` and `intrinsics_data` are single-precision.

Reply:
OpenCV performs calculations in 64F, probably having HAL which is fully compatible is a good idea.
And yes, it should take all the parameters as 64F, I'm fixing this.